### PR TITLE
Validate daemon mode

### PR DESCRIPTION
**Pull request by changweige**
We should give a explicit error if daemon mode argument is not legal.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -122,11 +133,16 @@ func SetStartupParameter(startupFlag *command.Args, cfg *Config) error {
 	cfg.PublicKeyFile = startupFlag.PublicKeyFile
 	cfg.ValidateSignature = startupFlag.ValidateSignature
 
+	if err := validateDaemonMode(startupFlag.DaemonMode); err != nil {
+		return errors.Wrapf(err, "invalidate daemon mode %s", startupFlag.DaemonMode)
```
Comment (imeoer) [inline config/config.go:137]: `invalid`

File: config/config.go [hunk 2]
```diff
@@ -86,22 +87,32 @@ type SnapshotterConfig struct {
...
+		return nil, errors.Wrapf(err, "load snapshotter configuration file %q", snapshotterConfigPath)
 	}
 	if err = tree.Unmarshal(config); err != nil {
-		return nil, errors.Wrapf(err, "failed to unmarshal snapshotter config file %q", snapshotterConfigPath)
+		return nil, errors.Wrapf(err, "unmarshal snapshotter configuration file %q", snapshotterConfigPath)
 	}
 	return &config, nil
 }
 
+func validateDaemonMode(mode string) error {
```
Comment (imeoer) [inline config/config.go:106]: How about

```
type DaemonMode string

func (dm DaemonMode) IsValid() error {
	switch dm {
	case DaemonModeMultiple, DaemonModeShared, DaemonModeNone, DaemonModePrefetch:
			return nil
	}
	return errdefs.ErrInvalidArgument
}
```

Comment (changweige) [reply config/config.go:106]: Nice code.

File: config/config.go (hunk 1 above)
Comment (changweige) [reply config/config.go]: fixed

File: config/config.go (hunk 2 above)
Comment (changweige) [reply config/config.go]: fixed

**Review by imeoer — APPROVED** [review]

**Pull request merged**